### PR TITLE
[importExternalReference] Bypasses certain security measures when generating a PDF or MD

### LINKED ISSUE #2753: [ImportExternalReference] BleepingComputer not importable due to Cloudflare protection
## Description

BleepingComputer not exportable in pdf format due to Cloudflare protection. The connector ImportExternalReference exports the Cloudflare portal in pdf format, rather than passing Cloudflare security to export the article page (see screenshot below)

## Environment

OCTI 6.3.4

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create an External Ref with:
    - https://www.bleepingcomputer.com/news/security/jpcert-shares-windows-event-log-tips-to-detect-ransomware-attacks/
2. Run the ImportExternalReference connector

## Expected Output

Have the BleepingComputer page in pdf format, not the Cloudflare portal checking that we're not a bot

## Actual Output

The pdf:

image

### PR DESCRIPTION
**Pull request by Megafredo**
### Proposed changes

* Adding more selectors for cookies consent
* If click cookie consent fails, we try to hide the unwanted elements
* Refacto process_playwright + Create a new browser context with a specified user agent to simulate a particular browser
* Remove useless variable env

### Related issues

* #2753 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality using different use cases
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

**Comment by SamuelHassine** [comment]
@Megafredo @helene-nguyen can we also put a user agent more standard using Playwright like the one we have in the previous version of the connector (Mozilla ...)?

File: internal-enrichment/import-external-reference/src/import-external-reference.py [hunk 1]
```diff
@@ -45,18 +45,92 @@ def __init__(self):
...
+            'button:has-text("allow")',
+        ]
+
+        for selector in selectors:
+            try:
+                page.locator(selector).first.click(timeout=1000, force=True)
+                found = True
+                break
+            except:
+                continue
```
Comment (flavienSindou) [inline internal-enrichment/import-external-reference/src/import-external-reference.py]: I think you could only except playwright error here and log message with debug level.

https://playwright.dev/python/docs/api/class-error [consulted on October 7th, 2024]

File: internal-enrichment/import-external-reference/src/import-external-reference.py [hunk 2]
```diff
@@ -179,28 +242,7 @@ def _process_external_reference(self, external_reference):
...
-                            try:
-                                page.locator(
-                                    'button:has-text("accept all")'
-                                ).first.click(timeout=1000, force=True)
-                                found = True
-                            except:
-                                pass
-                        if found:
-                            page.wait_for_timeout(2000)
+                        browser, page = self._process_playwright(p, url_to_import)
```
Comment (flavienSindou) [inline internal-enrichment/import-external-reference/src/import-external-reference.py:258]: Nice code splitting 👍.

**Review by flavienSindou — COMMENTED** [review]
Looks good to me 👍 
Just a remark on error handling.

**Review by flavienSindou — APPROVED** [review]
Code looks ok to me.

**Pull request merged**